### ISSUE TITLE
Version 1.0

### ISSUE THREAD

**Issue by glatard** [boutiques/boutiques#493]
Almost 40 tools are now published, and we didn't have a schema update that broke backward compatibility since release 0.5 on August 31, 2017. It might be a good time to have a release 1.0, that is, to prevent any API or schema update that would break backward compatibility. If we agree, the main action would be to document the Python API (!) and put it on readthedocs (see #280).
Thoughts?

**Comment by gkiar** [comment]
Only hesitation here is that we're currently talking about adding a possible KV-pair to the schema for BIDS apps, and while it'll be optional, it'd be nice to get that in to version 1.0 rather than quickly having to make another release. Otherwise, very pro!

Thoughts?

**Comment by glatard** [comment]
Agreed, we can add that KV-pair for 1.0. Although we *will* keep releasing every few months, it's just that we would commit to not breaking the schema.

**Comment by gkiar** [comment]
Agreed 😄

**Comment by glatard** [comment]
@camarasu, @ramou, any opinion?

**Comment by ramou** [comment]
I think we should break the schema as effectively as we can for the 1.0 release and then commit to no longer breaking it for an extended period. Change like that is necessary, so long as it is intermittent and planned, it is generally manageable. 

I would have gone with the painful break on KV pairs instead of the compatible version, but I wrote a few of them and looked at a few that were committed to the extent that I don't know what we went with.  Are we talking about keywords or other stuff? In any event, a major version change is a good time to break things and then promise not to do that again till the next major version change :D

**Comment by glatard** [comment]
Decision from the meeting: let's go for 1.0! We will go through the list of open issues before we release, and identify the ones we really want for 1.0.

**Comment by prioux** [comment]
I know this thread is already a year an a half old, but can I make a suggestion? The JSON schema need some cleanup and fixes. I was looking at it (because I'm crazy) and I found, for instance:

* there are duplicate IDs for some objects: "http://github.com/boutiques/boutiques-schema/description" and "http://github.com/boutiques/boutiques-schema/error-codes"
* some of these IDs, which are fake URLs, contain 'wrong' paths (if I can say something like that for a fake URL). E.g. "http://github.com/boutiques/boutiques-schema/output/required-inputs" and "http://github.com/boutiques/boutiques-schema/output/disabled-by-inputs" which are for *input* fields, despite the component `/output/` in them

Speaking of these IDs, I've always felt uncomfortable with their fake URL syntax. Can we consider changing them to something more meaningful, like maybe:

```
https://github.com/boutiques/boutiques/blob/master/tools/python/boutiques/schema#/path/to/object
https://github.com/boutiques/boutiques/tree/master/schema#/path/to/object

or simply:

/path/to/object

or even a URL with real content and fake anchor

https://raw.githubusercontent.com/boutiques/boutiques/master/tools/python/boutiques/schema/descriptor.schema.json#/path/to/object
```

I don't think this would break anything.... right?

**Comment by prioux** [comment]
Man GitHub turned all these IDs into links... which are dead of course.

**Comment by prioux** [comment]
Also, I'm hoping an ID in a JSON schema can contain the `#` character (notice I used it in my suggestions for new IDs)
